lumino-explorer-api/src/main/java/org/rif/lumino/explorer/managers/LuminoHubBalancingManager.java

```diff
@@ -0,0 +1,185 @@
+package org.rif.lumino.explorer.managers;
+
+import org.rif.lumino.explorer.boot.configuration.properties.LuminoHubProperties;
+import org.rif.lumino.explorer.exceptions.AlreadyExistException;
+import org.rif.lumino.explorer.exceptions.BadRequest;
+import org.rif.lumino.explorer.exceptions.MaxConnectionException;
+import org.rif.lumino.explorer.exceptions.NotFoundException;
+import org.rif.lumino.explorer.models.documents.LuminoHub;
+import org.rif.lumino.explorer.models.dto.LuminoHubDTO;
+import org.rif.lumino.explorer.repositories.LuminoHubRepository;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+@Service
+public class LuminoHubBalancingManager {
+
+  private final long INFINITE_CAPACITY = -1;
```

Comment:
this is not a blocker but infinite and -1 are completely oppositve

maybe we could use a Capacity class that has different values, or a long value, or a boolean "inifinteCapacity"

Reply:
what is your suggestion? i mean i put -1 since a negative capacity is not allowed so i put like a flag, but we can use a boolean if you want, i didn't add the boolean flag to avoid having to add another field to the class just for that specific case.